docs/oathkeeper/guides/websockets.mdx

```diff
@@ -0,0 +1,292 @@
+  kratos-selfservice-ui-node:
+    image: oryd/kratos-selfservice-ui-node:latest
+    environment:
+      - KRATOS_PUBLIC_URL=http://kratos:4433/
+      - KRATOS_BROWSER_URL=http://127.0.0.1:4433/
+    networks:
+      - intranet
+    ports:
+      - "4455:3000"
+    restart: on-failure
+  mailslurper:
+    image: oryd/mailslurper:latest-smtps
+    ports:
+      - '4436:4436'
+      - '4437:4437'
+    networks:
+      - intranet
+networks:
+  intranet:`}</CodeBlock>
+
+This example uses the following network architecture:
+
+- `4433` port is the public ("browser") API of Ory Kratos.
+- `4434` is the admin API of Ory Kratos.
+- `4455` is a port for the user interface implemented by the
+  [reference self-service UI](https://github.com/ory/kratos-selfservice-ui-node).
+- `8080` is a port of Ory Oathkeeper.
+
+Other ports and services are available only in the internal network.
+
+## Configure Ory Oathkeeper and Ory Kratos
+
+1. Create a `kratos` folder and fetch configuration files:
+
+<CodeBlock className="language-shell">{`mkdir kratos
+wget https://raw.githubusercontent.com/ory/kratos/${useLatestRelease(
+  'kratos'
+)}/contrib/quickstart/kratos/email-password/identity.schema.json -O kratos/identity.schema.json
+wget https://raw.githubusercontent.com/ory/kratos/${useLatestRelease(
+  'kratos'
+)}/contrib/quickstart/kratos/email-password/kratos.yml -O kratos/kratos.yml`}</CodeBlock>
+
+2. Create a `oathkeeper` folder and `oathkeeper/oathkeeper.yml` with the
+   following content:
+
+```yaml
+log:
+  level: debug
+  format: json
+
+serve:
+  proxy:
+    cors:
+      enabled: true
+      allowed_origins:
+        - http://127.0.0.1:8080
+      allowed_methods:
+        - POST
+        - GET
+        - PUT
+        - PATCH
+        - DELETE
+      allowed_headers:
+        - Authorization
+        - Content-Type
+      exposed_headers:
+        - Content-Type
+      allow_credentials: true
+      debug: true
+
+errors:
+  fallback:
+    - json
+
+  handlers:
+    redirect:
+      enabled: true
+      config:
+        to: http://127.0.0.1:4455/login
+        when:
+          - error:
+              - unauthorized
+              - forbidden
+            request:
+              header:
+                accept:
+                  - text/html
+    json:
+      enabled: true
+      config:
+        verbose: true
+
+access_rules:
+  matching_strategy: glob
+  repositories:
+    - file:///etc/config/oathkeeper/access-rules.yml
+
+authenticators:
+  anonymous:
+    enabled: true
+    config:
+      subject: guest
+
+  cookie_session:
+    enabled: true
+    config:
+      check_session_url: http://kratos:4433/sessions/whoami
+      preserve_path: true
+      extra_from: '@this'
+      subject_from: 'identity.id'
+      only:
+        - ory_kratos_session
+
+  noop:
+    enabled: true
+
+authorizers:
+  allow:
+    enabled: true
+
+mutators:
+  noop:
+    enabled: true
+```
+
+3. Create `oathkeeper/access-rules.yml` with the following content:
+
+```yaml
```

Comment:
Can you load these code snippets from the filesystem? Similar to what you have done with the other code samples?

````
 ```mdx-code-block title="ws/index.html"
 import index from '!!raw-loader!../../../code-examples/oathkeeper/ws/index.html'
 <CodeBlock language="html">{index}</CodeBlock>
 ```
````

Reply:
I can. BTW, do you know how can I use `${useLatestRelease}` hook with files on the filesystem? Or how can I update Ory services automatically? The reason is that right now there's no value to adding tests for current examples because they'll be on the same version since I pinned them to the latest release at the moment. In that case, I can't prove that everything works fine for the next release and we won't know if something is broken in examples.